### PR TITLE
[MRG+1] Bump minimum statsmodels

### LINKED ISSUE #332: 'SARIMAX' Object has no attribute '_k_trend' when imported from a pickle file
I have created a pipeline using following code 
`pipe = Pipeline([
    ('boxcox', BoxCoxEndogTransformer(lmbda2=1e-6)),
    ('arima', pmd.AutoARIMA(trace=True,
                           suppress_warnings=True,
                           m=12,
                           stepwise=True))
])`

and fitted it like this 
`fitted = pipe.fit(train)`
Afterwards I created a pickle dump (this part was done in google colabs, pmdarima-1.6.0)
Then I created a local application perform predictions locally, however I am getting this error 
`AttributeError: 'SARIMAX' object has no attribute '_k_trend'`
Can anyone explain me what have I done wrong here ?

### PR DESCRIPTION
**Pull request by aaronreidsmith**
# Description

Statsmodels 0.11.0 introduces the `_k_trend` attribute, which will fail if trying to un-pickle a model created on an older statsmodels. Easiest solution is to just bump minimum statsmodels from 0.10.2 -> 0.11.0

Fixes #332 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Pickled a model using 0.11.0:

```python
import pickle

import pmdarima as pm
from pmdarima.model_selection import train_test_split
import statsmodels


assert statsmodels.__version__ == '0.11.0'


# Load/split your data
y = pm.datasets.load_wineind()
train, test = train_test_split(y, train_size=150)

# Fit your model
model = pm.auto_arima(train, seasonal=True, m=12)

with open('/Users/asmith/Desktop/model.pickle', 'wb') as file:
    pickle.dump(model, file)
```

Unpickled using 0.11.1, just to be sure:

```python
import pickle

import matplotlib.pyplot as plt
import numpy as np
import pmdarima as pm
from pmdarima.model_selection import train_test_split
import statsmodels


assert statsmodels.__version__ == '0.11.1'

y = pm.datasets.load_wineind()
train, test = train_test_split(y, train_size=150)

model = None
with open('/Users/asmith/Desktop/model.pickle', 'rb') as file:
    model = pickle.load(file)

# make your forecasts
forecasts = model.predict(test.shape[0])  # predict N steps into the future

# # Visualize the forecasts (blue=train, green=forecasts)
x = np.arange(y.shape[0])
plt.plot(x[:150], train, c='blue')
plt.plot(x[150:], forecasts, c='green')
plt.show()
```

# Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pmdarima/arima/tests/test_arima.py [hunk 1]
```diff
@@ -768,7 +768,7 @@ def test_warn_for_large_differences():
 def test_warn_for_stepwise_and_parallel():
     with pytest.warns(UserWarning):
         auto_arima(lynx, suppress_warnings=False, d=1,  # noqa: F841
-                       error_action='ignore', stepwise=True, n_jobs=2)
+                   error_action='ignore', stepwise=True, n_jobs=2)
```
Comment (aaronreidsmith) [inline pmdarima/arima/tests/test_arima.py:771]: Not sure why the linter complained about this... I did not edit it until the linter complained.

Comment (aaronreidsmith) [reply pmdarima/arima/tests/test_arima.py:771]: Flake8 had a release a [few days ago](https://flake8.pycqa.org/en/latest/release-notes/index.html), so maybe something with that

Comment (tgsmith61591) [reply pmdarima/arima/tests/test_arima.py:771]: Ha, that's funny. I found the same thing in another codebase I manage. `flake8` now complains if you have an F-string with no formatting fields in it. Good catch, I'm sure that probably fit the other fella who submitted a PR

**Comment by aaronreidsmith** [comment]
This PR turned out to be bigger than bumping the version. I also had to do the following:

1. Rip out regression tests
    - statsmodels 0.11.0 is no longer compatible with legacy `pyramid-arima`. Since it has been deprecated for 18+ months, this is not a huge concern
2. Finally force me to figure out our pypy issues
    - Had to use `--no-use-pep517` to build some of our dependencies
    - Had to bump us to use `pypy3 7.3.0` (see alkaline-ml/pmdarima-docker#4)

**Pull request merged**